aws_xray_sdk/ext/httplib/patch.py

```diff
@@ -22,8 +22,33 @@
 
 _XRAY_PROP = '_xray_prop'
 _XRay_Data = namedtuple('xray_data', ['method', 'host', 'url'])
+_XRay_Ignore = namedtuple('xray_ignore', ['subclass', 'hostname', 'urls'])
 # A flag indicates whether this module is X-Ray patched or not
 PATCH_FLAG = '__xray_patched'
+# Calls that should be ignored
+_XRAY_IGNORE = set()
+
+
+def add_ignored(subclass=None, hostname=None, urls=None):
+    global _XRAY_IGNORE
+    if subclass is not None or hostname is not None or urls is not None:
+        urls = urls if urls is None else tuple(urls)
+        _XRAY_IGNORE.add(_XRay_Ignore(subclass=subclass, hostname=hostname, urls=urls))
+
+
+def reset_ignored():
+    global _XRAY_IGNORE
+    _XRAY_IGNORE.clear()
+    _ignored_add_default()
+
+
+def _ignored_add_default():
+    # skip httplib tracing for SDK built-in centralized sampling pollers
+    add_ignored(subclass='botocore.awsrequest.AWSHTTPConnection', urls=['/GetSamplingRules', '/SamplingTargets'])
```

Comment:
I'm conflicted here - I know that using Python's built-in string matching as we are now will be faster than using `fnmatch`, but it would be awkward to not use this ignore mechanism and special-case /GetSamplingRules and /SamplingTargets. 

I guess the added latency is kinda peanuts compared to the actual network request, so it's probably ok, but what do you think @srprash?

Reply:
I feel using `fnmatch` for ignoring /GetSamplingRules and /SamplingTarget is okay since it fits well in the overall mechanism to ignore any URL. I'm not really aware of the latency of `fnmatch` but my guess is that special casing the sampling urls and then matching the user urls would be roughly equivalent and won't cause much of a difference here.